### DISCUSSION TITLE
Code Review IP-93

### DISCUSSION THREAD

**Discussion by MaksGovor** [HowProgrammingWorks/LiveQA#81] [Code review]
- Backend: https://github.com/Intelligent-Cloud-System/ics-backend/pull/18
- Frontend: https://github.com/Intelligent-Cloud-System/ics-frontend/pull/10

Примітка: у Frontend частині є папка `/src/clients/` в якій знаходиться автозгенерований код `openAPI` (command to generate: `yarn generate-clients`) , котрий не потребує review

Також має прийти запрошення на outside collaborator в дані репозиторії
Над проектом працювали студенти ІП-93:

| Contributor                                         | Telegram                                | Gmail                                                    |
| :-------------------------------------------------: |:---------------------------------------:| :-------------------------------------------------------:|
| [Говоруха Максим](https://github.com/MaksGovor)      | [Maksgovorrr](https://t.me/Maksgovorrr) | [maksgovruha@gmail.com](mailto:maksgovruha@gmail.com)    |
| [Урин Дмитро](https://github.com/tedi4t)            | [tedi4t](https://t.me/tedi4t)           | [tedi4t@gmail.com](mailto:tedi4t@gmail.com)              |

**Comment by tshemsedinov** [comment]
- https://github.com/Intelligent-Cloud-System/ics-backend/pull/18
- https://github.com/Intelligent-Cloud-System/ics-frontend/pull/10

![photo_2022-05-19_21-35-45](https://user-images.githubusercontent.com/4405297/172841109-856bbb25-183a-4a78-93e4-8686cd725986.jpg)

**Comment by tedi4t** [comment]
Hi. We have added emitting WebSockets from back-end to front-end. We use it to emit events when something is updated on our back-end (for example, when user adds a new file to organization folder, all organization members receive event that files list is updated and then they decide whether they need to do a query to update content). In future, we will likely rewrite our server for only WS, but for now we use our server as an SDK thanks to the Swagger and we are waiting when NestJS or other framework will introduce OpenAPI for WS. Maybe OpenAPI for WS will be our next project :)

We have also added an algorithm to uniquely generate an avatar (Julia set fractal algorithm) for every user based on his email. Then, this icon is saved on S3 Bucket and then user can get this icon via Signed Url that he gets from our back-end. Below you can see an example of this avatar.

Also, I would like to say that [Uryn Dmytro](https://github.com/tedi4t) has contributed in 2022, but git email on his local machine is wrong, that's why GitHub doesn't see his work in "Contributors" tab of "Insights", but you can see it in "Commits" tab.

Besides, [Uryn Dmytro](https://github.com/tedi4t) and [Maxim Govoruha](https://github.com/MaksGovor) study in two subjects where you are a teacher: "Технології розроблення Back-end", "Гнучкі методи програмування". [Grybenko Yegor](https://github.com/YAGoOaR) studies only "Гнучкі методи програмування".

![avatar](https://user-images.githubusercontent.com/44302526/175292551-41e7c8f8-92d5-4cda-a3da-321f63bac3ca.png)
